### DISCUSSION TITLE
Support gitrevset in args that accept commits

### DISCUSSION THREAD

**Discussion by bcongdon** [arxanas/git-branchless#157] [Ideas]
It looks like git-branchless already has a dependency on https://github.com/quark-zju/gitrevset

Allowing revset expressions to be used as values to args/flags that accept commits would be nice.

e.g. being able to do `git hide descendants(abcd1234)` would be handy

**Comment by arxanas** [comment]
That's the plan!

In my assessment, we would be able to reuse the parser, but probably not the evaluator (see https://github.com/quark-zju/gitrevset/issues/1, https://github.com/quark-zju/gitrevset/issues/2). Originally, I was trying to integrate directly with `gitrevset`, but their models for updating the DAG are rather different. For example, we have an event log which tells us which commits are obsolete, while the `gitrevset` library attempts to infer somehow else (from the reflog?).

The place to add a call to evaluate a revset would be in `Dag::resolve_commits`:

https://github.com/arxanas/git-branchless/blob/279e754d14f03491cb3494c73e483f6c6a1cb23e/src/git/dag.rs#L531-L560

It'll require some refactoring since current it operates on a list of OIDs instead of individual OIDs (for historical reasons), and it returns a `Vec<Commit>`. Instead, it should accept a revset expression and return a `CommitSet`. If you plan to add revset support yourself, then open an issue and assign it to yourself, so that I don't duplicate your work at some point in the future.
